test/common/tls/context_impl_test.cc

```diff
@@ -1421,16 +1421,12 @@ TEST_F(ClientContextConfigImplTest, NonP256EcdsaCert) {
                             *tls_context.mutable_common_tls_context()->add_tls_certificates());
   auto client_context_config = *ClientContextConfigImpl::create(tls_context, factory_context_);
   Stats::IsolatedStoreImpl store;
-  EXPECT_THAT(manager_.createSslClientContext(*store.rootScope(), *client_context_config)
-                  .status()
-                  .message(),
-              testing::ContainsRegex(
-                  "Failed to load certificate chain from .*selfsigned_ecdsa_p384_cert.pem, "
-                  "only P-256 ECDSA certificates are supported"));
+  auto context = *manager_.createSslClientContext(*store.rootScope(), *client_context_config);
+  auto cleanup = cleanUpHelper(context);
```

Comment:
I just noticed that the existing P-256 success test was broken in a refactoring to remove exceptions (I'll PR the fix soon. Please apply that fix to this new code
```suggestion
  auto context_or = manager_.createSslClientContext(*store.rootScope(), *client_context_config);
  EXPECT_TRUE(context_or.ok());
  auto cleanup = cleanUpHelper(*context_or);
```